pallets/system/src/submitted_data.rs

```diff
@@ -162,6 +191,18 @@ where
 	proof(submitted_data, data_index, Rc::clone(&metrics))
 }
 
+/// Extracts the AppId from extrinsic, if it is signed, returns None otherwise.
+pub fn extract_app_id<AE>(extrinsic: &OpaqueExtrinsic) -> Option<AppId>
```

Comment:
Can `extract_app_id_and_inspect` and `extract_app_id` be merged into one function?

Reply:
Yes, it's basically just an unnecessary step, but I did it to match other similar functions.